### ISSUE TITLE
Feature request. Don't start a countdown until a user approves that he is ready to work

### ISSUE THREAD

**Issue by matchish** [hovancik/stretchly#522]
### Prerequisites

- [x] I'm using latest version: v 0.21.0
- [x] I've checked [Known issues](https://github.com/hovancik/stretchly#known-issues)

### Description

When a break starts I don't wait for break end near laptop, and often break is end before I come back, and a next working interval can be shorter for me then expected.
Feature request. Don't start a countdown until a user approves that he is ready to work.

Operating system: Any

### Steps to Reproduce

1. Start break 5 min (work interval 20min)
2. Go to a street for 10 min
3. Come back and start to work

**Expected behavior:** Next interval should be 20 min

**Actual behavior:** Working interval will be 10 min until next break 

**Reproduces how often:** 100%

**Comment by hovancik** [comment]
Hi there and thanks for the idea.

What you can do right now:
- make sure you have stretchly 0.21.1 released today  
- make sure you have "monitor system idle time for natural breaks" enabled
- edit "Natural breaks inactivity time" via [config file](https://github.com/hovancik/stretchly#natural-breaks-inactivity-time) / contributor settings to 1 minute (60000ms if my math is correct) 

When you go to break and come back 5 minutes after stretchly's break has ended, stretchly's timer will reset upon your arrival.

**Comment by yj2303** [comment]
I would like to contribute in this issue.

**Comment by hovancik** [comment]
Hi @yj2303 and thanks for your interest ;] 

How do you image this would work from UX perspective?

**Comment by yj2303** [comment]
We can solve this issue by using face detection for resetting the time . If a person takes break before the estimated time then he must have to switch off the system or put it on sleep then we will stop the alarm and as soon as he reopens it alarm will reset itself. I think this idea should be followed . Or you can guide me for the proceedings and I will follow up the same.

**Comment by matchish** [comment]
The issue is if I come back later then break timer finished then my next work interval will be less then defined. It does not about take a break before the estimated time.
And this one doesn't work in my case (maybe because I often have some background tasks) https://github.com/hovancik/stretchly/issues/522#issuecomment-575181266

**Comment by yj2303** [comment]
Okay, I got it that we have to reset the time on arrival of the person. It is quite interesting task. kindly assign it to me.

**Comment by hovancik** [comment]
@matchish did you try the solution that I recommended with inactivity time? 

@yj2303 there's no need to assign, yet ;] software development is more then coding, we first need to figure out the solution, validate and then we can start thinking about implementation

stretchly has strict rules for coding, so I won't accept solution that was not first discusses inside of issue. That's because codebase is quite complicated already ;] 

so for this specific issue: 
- problem is when user with 10 minutes interval comes ie 3 minutes after break, thus having this interval "shortened" to 7 minutes
- user would like pause interval countdown until they are ready to work
- we already have a functionality that pause countdown when user is idle, so we should probably use it

Anyway, we need to come up with solution that will serve majority of users, not only one. I think there were talks in another issues about similar problems, we should try to find them. 

Then we can generalize solution.

**Comment by matchish** [comment]
@hovancik Yes, I tried but it doesn't work for some reason in my case. How do you  check that system idle?

**Comment by hovancik** [comment]
@matchish prereqs: 

```
    "naturalBreaks": true,
    "naturalBreaksInactivityResetTime": 60000,
```
I would recommend to turn off stretchly before editing config file. 

What I do then is work normally, let's say I am 5 minutes to microbreak and then leave computer for minute and something. When I come back, I am `10 minutes to microbreak`

**Comment by matchish** [comment]
> I would recommend to turn off stretchly before editing config file.

@hovancik Now it works. Thanks.

**Comment by matchish** [comment]
My issue is solved. But if someone interested in manual confirmation we can do it in this way
Popup with countdown is visible until no mouse activity.
After countdown is finished, color of popup changes and there is sound alarm

**Comment by phillippschmedt** [comment]
I am still looking for a feature that enables this in stretchly. 

Note: I am using full-screen mode. 

How about:
* Add an option to enable user confirmation to end a break
* This will make the window stay open even when the break is over.
* Put a button "Finish break" which will close the window and start the next timer

This would meet my personal requirements.

Any opinions on this?

**Comment by xeruf** [comment]
Since Stretchly can already track activity, why not simply keep the break window open until the mouse is moved or any key is pressed rather than having a specific button?

**Comment by hovancik** [comment]
If one using using activity tracking, breaks will be resumed after they start activity again. 

In default behaviour, where Long Break is 5 minutes, if user takes breaks 7 minutes: breaks will be paused after 5 minutes of inactivity. Therefore they will be resumed after user is back.

**Comment by xeruf** [comment]
the problem is that changing the length of the long break does not change the reset timer, I have to manually edit the config for that, which is not very handy.

**Comment by hovancik** [comment]
Well, it's not only connected to this, so I prefer to leave this to user.

**Comment by MM-John** [comment]
My issue is that I do things ON the computer during my break. So the idle timer workaround won't work for me.

As for UI: Preferences already has "Enable Postponement for * Breaks". 

So a "Prompt to start next work session" seems suitable. Or an "Automatically start next work session", default checked on, might work.

Also, maybe, in the break window, an "Extend Break" option. Maybe with a time.

**Comment by r3a1d3a1** [comment]
[phillippschmedt](https://github.com/phillippschmedt)'s idea (quoted below) would be perfect for my case as well:
> * Add an option to enable user confirmation to end a break
> * This will make the window stay open even when the break is over.

(My system is always doing stuff that makes automatic idle detection a moot point.)

This implementation shouldn't take more than 2hrs of someone familiar with the codebase. Right?
